source/string.c

```diff
@@ -57,11 +65,25 @@ void aws_string_destroy_secure(struct aws_string *str) {
 }
 
 int aws_string_compare(const struct aws_string *a, const struct aws_string *b) {
+    AWS_PRECONDITION(aws_string_is_valid(a));
+    AWS_PRECONDITION(aws_string_is_valid(b));
+    if (a == b) {
+        return 0; /* strings identical */
+    }
+    if (a == NULL) {
+        return -1;
+    }
+    if (b == NULL) {
+        return 1;
+    }
```

Comment:
@ColdenCullen Again, should I consider the case where `a` or `b` could be equal to `NULL` or just replace `AWS_PRECONDITION` with `AWS_FALTAL_ASSERT`?

Reply:
You don't need to change anything here but the preconditions, if they're both null the first check will trigger.